File: 프로그래머스/lv2/131127. 할인 행사/할인 행사.js
function solution(want, number, discount) {
    var answer = 0;
    var list = {};
    for(let i = 0;i<discount.length-9;i++){
        var total = 0;
        var check = false;
        for(let i =0;i<want.length;i++){
            list[want[i]] = number[i];
        }
        for(let j = i;j<i+10;j++){
            if(list.hasOwnProperty(discount[j])){
                list[discount[j]] -= 1;
            }
        }
        for(const key in list){
            if(list[key] !== 0){
                check = true;
                break;
            }
        }
        if(!check){
            answer++;
        }
    }
    return answer;
}

File: 프로그래머스/2/131127. 할인 행사/할인 행사.cpp
#include <string>
#include <vector>
#include <unordered_map>
#include <iostream>
using namespace std;

int solution(vector<string> want, vector<int> number, vector<string> discount) {
    int answer = 0;
    unordered_map<string,int> mymap;
    int num = 0;
    for(int i = 0;i<want.size();i++){
        mymap[want[i]] = number[i];
        num += number[i];
    }
    for(int i = 0;i<discount.size()-9;i++){
        unordered_map<string,int> comp_map = mymap;
        bool flag = false;
        for(int j = i;j<i+10;j++){
            if(comp_map.find(discount[j]) != comp_map.end()){
                comp_map[discount[j]]--;
                if(comp_map[discount[j]] < 0){
                    flag = true;
                    break;
                }
            }
            else{
                flag = true;
                break;
            }
        }
        if(!flag){
            answer++;
        }
    }
    return answer;
}